tensorboard/webapp/app_routing/dirty_updates_registry_module.ts

```diff
@@ -0,0 +1,78 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {Inject, ModuleWithProviders, NgModule, Optional} from '@angular/core';
+
+import {EXPS_WITH_DIRTY_UPDATES_TOKEN} from './dirty_updates_registry_types';
+import {DirtyUpdates} from './types';
+
+@NgModule()
+export class DirtyUpdatesRegistryModule {
+  private readonly dirtyUpdates: DirtyUpdates | null = null;
+
+  constructor(
+    @Optional() @Inject(EXPS_WITH_DIRTY_UPDATES_TOKEN) eidsList: string[][]
+  ) {
+    if (!eidsList) {
+      return;
+    }
+    const experimentIds: string[] = [];
+    for (const eids of eidsList) {
+      for (const eid of eids) {
+        experimentIds.push(eid);
+      }
+    }
```

Comment:
I suspect you will not require this code block after the change but in the future, try `[[1], 2].flat()` instead :)

Reply:
Second suspicion is also correct!